source/server/configuration_impl.h

```diff
@@ -130,9 +185,16 @@ class MainImpl : Logger::Loggable<Logger::Id::config>, public Main {
     std::list<NetworkFilterFactoryCb> filter_factories_;
   };
 
-  static std::list<NetworkFilterConfigFactory*>& filterConfigFactories() {
-    static std::list<NetworkFilterConfigFactory*> filter_config_factories;
-    return filter_config_factories;
+  static std::unordered_map<std::string, NetworkFilterConfigFactory*>& filterConfigFactories() {
+    static std::unordered_map<std::string, NetworkFilterConfigFactory*>* filter_config_factories =
```

Comment:
Yeah, the static initialization order fiasco :) See https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use. We have a section on this now in https://github.com/lyft/envoy/blob/master/STYLE.md.